src/paginationCalls/pageDescriptor.ts

```diff
@@ -103,6 +116,107 @@ export class PageDescriptor implements Descriptor {
     return stream;
   }
 
+  /**
+   * Create an async iterable which can be recursively called for data on-demand.
+   */
+  asyncIterate(
+    apiCall: GaxCall,
+    request: RequestType,
+    options: CallSettings
+  ): AsyncIterable<{} | undefined> {
+    const iterable = this.createIterator(options);
+    const funcPromise =
+      typeof apiCall === 'function' ? Promise.resolve(apiCall) : apiCall;
+    funcPromise
+      .then((func: GaxCall) => {
+        this.resolveParams(request, func, options);
+      })
+      .catch(error => {
+        throw new Error(error);
+      });
+    return iterable;
+  }
+
+  createIterator(options: CallSettings): AsyncIterable<{} | undefined> {
+    const responsePageTokenFieldName = this.responsePageTokenField;
+    const requestPageTokenFieldName = this.requestPageTokenField;
+    let resolveRequest = this.resolveRequest;
+    let resolveFunction = this.resolveFunction;
+    const asyncIterable = {
+      [Symbol.asyncIterator]() {
+        const funcPromise = new Promise((resolve, reject) => {
+          resolveFunction = resolve;
```

Comment:
you assign `resolveFunction` and `resolveRequest` (local variables defined in lines 143 and 144) but you never assign `this.resolveRequest` and `this.resolveFunction`. Maybe you don't need private members at all and these two local variables in the syntactic closure will work for us here?

Reply:
I think `this.resolveRequest` and `this.resloveFunction` is initialized in constructor and assigned in `resolveParams` function [line 214]? Correct me if I'm wrong, so these two local variables are for passing the private members `this.resolveRequest` and `this.resolveFunction` to the iterator.